src/bin/main.rs

```diff
@@ -387,16 +387,9 @@ fn format_and_emit_report<T: Write>(session: &mut Session<'_, T>, input: Input)
 }
 
 fn should_print_with_colors<T: Write>(session: &mut Session<'_, T>) -> bool {
-    match term::stderr() {
-        Some(ref t)
-            if session.config.color().use_colored_tty()
+    matches!(term::stderr(), Some(ref t) if session.config.color().use_colored_tty()
                 && t.supports_color()
-                && t.supports_attr(term::Attr::Bold) =>
-        {
-            true
-        }
-        _ => false,
-    }
+                && t.supports_attr(term::Attr::Bold))
 }
```

Comment:
#5400 is using `is_some_and`, which I think it a little clearer than the `matches!` that you're using here.

```rust
    term::stderr().is_some_and(|t| {
        session.config.color().use_colored_tty()
            && t.supports_color()
            && t.supports_attr(term::Attr::Bold)
    })
```

Reply:
fixed